program/program/Program.cs

```diff
@@ -0,0 +1,246 @@
+            }
+        }
+
+        static void Menu()
+        {
+            Console.Clear();
+            Console.WriteLine("Welcome to Phone Book Application!\n");
+            Console.WriteLine("\tMenu");
+            Console.WriteLine("\t1. Show all phone book");
+            Console.WriteLine("\t2. Create phone record");
+            Console.WriteLine("\t3. Search by name");
+            Console.WriteLine("\t4. Search by phone number");
+            Console.WriteLine("\t5. Update user");
+            Console.WriteLine("\t0. Exit");
+
+            ConsoleKeyInfo ck = Console.ReadKey();
+
+            switch (ck.Key)
+            {
+                case ConsoleKey.D1:
+                case ConsoleKey.NumPad1:
+                    ShowAllNumbers();
+                    break;
+                case ConsoleKey.D2:
+                case ConsoleKey.NumPad2:
+                    CreatePhoneNumber();
+                    break;
+                case ConsoleKey.D3:
+                case ConsoleKey.NumPad3:
+                    SearchByName();
+                    break;
+                case ConsoleKey.D4:
+                case ConsoleKey.NumPad4:
+                    SearchByPhoneNumber();
+                    break;
+                case ConsoleKey.D5:
+                case ConsoleKey.NumPad5:
+                    UpdateUser();
+                    break;
+                case ConsoleKey.D0:                    
+                case ConsoleKey.NumPad0:
+                    Exit();
+                    break;
+            }
+        }
+
+        private static void SearchByName()
+        {
+            Console.Clear();
+            Console.WriteLine("Search condition - if first/last name contains search term you will see it");
+
+            Console.WriteLine("Enter Search Term...");
+            var searchTerm = Console.ReadLine();
+
+            bool found = false;
+            foreach (var (firstName, lastName, phoneNumber) in ReadPhoneBook())
+            {
+                if (firstName.Contains(searchTerm, StringComparison.OrdinalIgnoreCase) ||
+                    lastName.Contains(searchTerm, StringComparison.OrdinalIgnoreCase))
+                {
+                    Console.WriteLine($"Found user {firstName} {lastName} with phone {phoneNumber}");
+                    found = true;
+                }
+            }
+
+            if (!found)
+            {
+                Console.WriteLine("Not such users");
+            }
+
+            Wait();
+        }
+
+        private static void SearchByPhoneNumber()
+        {
+            Console.Clear();
+            Console.WriteLine("Search by the name.");
+            Console.WriteLine("Pls enter the phone nuber");
+
+            var searchTerm = Console.ReadLine();
+
+            bool found = false;
+            foreach (var (firstName, lastName, phoneNumber) in ReadPhoneBook())
+            {
+                if (phoneNumber.Contains(searchTerm, StringComparison.OrdinalIgnoreCase))
```

Comment:
not critical, you do not need StringComparison.OrdinalIgnoreCase when comparing number